src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,74 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        // exit immediately if the target file already exists
+        File file = new File(toFileName);
+        if (file.exists()) {
+            return;
+        }
+        String dataFromFile = readFromFile(fromFileName);
+        String report = generateReport(dataFromFile);
+        writeToFile(toFileName, report);
+    }
+
+    private String readFromFile(String fileName) {
+        StringBuilder builder = new StringBuilder();
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(fileName));
+            // I read data symbol by symbol to preserve the original formatting
+            int symbol = reader.read();
+            while (symbol != -1) {
+                builder.append((char) symbol);
+                symbol = reader.read();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read from file" + fileName, e);
+        }
+
+        return builder.toString();
+    }
+
+    private String generateReport(String data) {
+        int supply = 0;
+        int buy = 0;
+        String[] entries = data.split("\r\n?|\n"); // split by line separator
+
+        for (String entry: entries) {
+            String[] values = entry.split(",");
+            switch (values[0]) {
```

Comment:
Create constants for 0 and 1
```suggestion
            switch (values[OPERATION_INDEX]) {
```